boards/arm/nrf9160_pca10090/Kconfig.board

```diff
@@ -4,6 +4,14 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+if SOC_NRF9160_SICA
+
 config BOARD_NRF9160_PCA10090
 	bool "nRF9160 PCA10090"
-	depends on SOC_NRF9160_SICA
+	depends on !TRUSTED_EXECUTION_NONSECURE
```

Comment:
You could skip this dependence (although it is correct). It is not useful, I think. The default value will be false, anyways.

Reply:
My intention was to keep only one choice available. If that condition is omitted, then in case of non-secure firmware both the secure and non-secure board versions are shown in menuconfig. Although the right one will be selected, I'd rather avoid that.

Reply:
Could there be a choice?

Reply:
No, there is no choice; when building for secure the board is nrf9160_pca10090, when building for non-secure the board is nrf9160_pca10090ns. So removing the dependency will let show a choice that is wrong.